src/thread_pool.rs

```diff
@@ -927,25 +925,10 @@ impl WorkerThread {
         &mut *self.queue.get()
     }
 
-    /// Sets `self` as the worker thread index for the current thread.
-    /// This is done during worker thread startup.
-    ///
-    /// # Safety
-    ///
-    /// This must be called only once per thread.
-    unsafe fn set_current(&self) {
-        WORKER_THREAD_STATE.with(|t| {
-            assert!(t.get().is_null());
-            t.set(self);
-        });
-    }
-
-    /// Gets the `WorkerThread` for the current thread; returns NULL if this is
-    /// not a worker thread. This pointer is valid anywhere on the current
-    /// thread.
-    #[inline]
-    pub fn current() -> *const WorkerThread {
-        WORKER_THREAD_STATE.with(Cell::get)
+    /// Acquires a reference to the `WorkerThread` for the current thread.
+    /// This will be `None` if the current thread is not a worker thread.
+    pub fn with<R>(f: impl FnOnce(Option<&Self>) -> R) -> R {
+        WORKER_THREAD_STATE.with(|worker_thread| f(worker_thread.get()))
```

Comment:
Much better. This bit was pure rayon and shows it's age. It's from before OnceCell was stabilized.